eng/submit-source-build-release-pr.sh

```diff
@@ -135,7 +150,43 @@ cat "$global_json_path" \
     | tee "$global_json_path.new"
 mv "$global_json_path.new" "$global_json_path"
 
-sed -i "s#<PrivateSourceBuiltArtifactsPackageVersion>.*</PrivateSourceBuiltArtifactsPackageVersion>#<PrivateSourceBuiltArtifactsPackageVersion>$sdk_version</PrivateSourceBuiltArtifactsPackageVersion>#" $versions_props_path
+# Function to modify the URL in an XML file
+# Arguments:
+#   1. XML file path
+#   2. Element name
+#   3. Replacement value
+function modify_url_in_xml() {
+  local xml_file="$1"
+  local element_name="$2"
+  local replacement_value="$3"
+
+  # Fetch the URL inside the element
+  local url=$(grep -oP "<$element_name>\K.*?(?=<\/$element_name>)" "$xml_file")
+
+  # Replace the very last part of the URL with the provided value
+  local new_url=$(echo "$url" | sed "s|/[^/]*$|/$replacement_value|")
+
+  # Update the XML file with the modified URL
+  sed -i "s|$url|$new_url|" "$xml_file"
+
+  echo "Modified URL: $new_url"
+}
+
+if [[ $sdk_version == "6"* ]]; then
+        sed -i "s#<PrivateSourceBuiltArtifactsPackageVersion>.*</PrivateSourceBuiltArtifactsPackageVersion>#<PrivateSourceBuiltArtifactsPackageVersion>$sdk_version</PrivateSourceBuiltArtifactsPackageVersion>#" $versions_props_path
+elif [[ $sdk_version == "7"* ]]; then
+        sed -i "s#<PrivateSourceBuiltArtifactsPackageVersion>.*</PrivateSourceBuiltArtifactsPackageVersion>#<PrivateSourceBuiltArtifactsPackageVersion>$sdk_version</PrivateSourceBuiltArtifactsPackageVersion>#" $versions_props_path
+        sed -i "s#<PrivateSourceBuiltSDKVersion>.*</PrivateSourceBuiltSDKVersion>#<PrivateSourceBuiltSDKVersion>$sdk_version</PrivateSourceBuiltSDKVersion>#" $versions_props_path
+elif [[ $sdk_version == "8"* ]]; then
+        modify_url_in_xml "$versions_props_path" "PrivateSourceBuiltArtifactsUrl" "$resolved_source_built_artifacts_file_name"
+        modify_url_in_xml "$versions_props_path" "PrivateSourceBuiltSdkUrl_CentOS8Stream" "$resolved_sdk_artifact_file_name"
+else
+        echo "Unexpected SDK version!"
```

Comment:
1) It's good to write also the value that led to the error, e.g: "Version '$sdk_version' is not supported"
2) Errors should go to STDERR (standard error stream), i.e., `echo "..." >&2`.

Reply:
Premek asked me to remove this part of the code (else).

So, the final version would be like this:

if 6 or 7
	# todo


if 7
 	# todo
elif not 6 (targeting version 8 and subsequent ones)
	# todo

Reply:
This seems cleaner:

```
if 6 or 7
  # todo
  if 7
    # todo
  fi
else
  # todo
fi
```

Reply:
It should be nested:
```
if 6 or 7:
  ...
  if 7:
    ...
else
  ...
```